### PR TITLE
Fix crash in /randset

### PR DESCRIPTION
**Pull request by mia-pi-git**
Was caused by the mon not having a learnset for this gen.

### CODE REVIEW

File: server/chat-plugins/random-battles.ts [hunk 1]
```diff
@@ -824,6 +824,10 @@ export const commands: ChatCommands = {
 		if (!species.exists) {
 			throw new Chat.ErrorMessage(`${pokemonString} is not a valid Pokémon for that format.`);
 		}
+		const dex = Dex.forFormat(format);
+		if (!dex.getLearnsetData(species.id).learnset) {
+			return this.errorReply(`Learnsets for ${species.name} in ${format.name} not found.`);
```
Comment (AnnikaCodes) [inline server/chat-plugins/random-battles.ts:829]:
```suggestion
			throw new Chat.ErrorMessage(`Learnsets for ${species.name} in ${format.name} not found.`);
```

**Review by AnnikaCodes — DISMISSED** [review]
Looks good, thanks!

**Comment by mia-pi-git** [comment]
> D130CAB1-E34D-4239-9C1E-32042304C9D9
> It’s not a matter of which gen the Pokémon is in. Seems like someone tried to use `/randset` on a Pokestar Pokémon, which have no learnset in any gen.

Yes, i'm aware. That's what i'm trying to fix, by adding a more general solution to check if they have a learnset in the target format at all.

**Comment by AnnikaCodes** [comment]
Both checks should use the modded dex.

**Review by AnnikaCodes — APPROVED** [review]
Looks good, thanks!

**Pull request merged**